src/comments.js

```diff
@@ -0,0 +1,53 @@
+import ADD_COMMENT from './actions' ;
+import REMOVE_COMMENT from './actions' ;
+import EDIT_COMMENT from './actions';
+import THUMB_UP_COMMENT from './actions' ;
+import THUMB_DOWN_COMMENT from './actions' ;
+
+function comments(state = [], action) {
+  switch(action.type) {
+    case ADD_COMMENT:
+      return [{  
+            id: action.id,
+            text: action.text,
+            votes: 0
+          }
+          , ...state.comments];
```

Comment:
your state doesn't have an comments value because as a default value you added an empty array